src/features/validation/expressionValidation/ExpressionValidation.tsx

```diff
@@ -91,6 +91,17 @@ function IndividualExpressionValidation({ dataType }: { dataType: string }) {
                 valueArguments,
               },
             );
+            const evaluatedMessage = evalExpr(
+              validationDef.message as ExprValToActualOrExpr<ExprVal.String>,
```

Comment:
I would have set this type (`ExprValToActualOrExpr<ExprVal.String>`) in `IExpressionValidation` instead, so that you don't have to cast the type.